WalletWasabi.Fluent/Models/Wallets/WalletModel.cs

```diff
@@ -19,6 +19,7 @@ namespace WalletWasabi.Fluent.Models.Wallets;
 public partial class WalletModel : ReactiveObject, IWalletModel
 {
 	private readonly TransactionHistoryBuilder _historyBuilder;
+	private readonly Lazy<IWalletBalancesModel> _balances;
```

Comment:
@ichthus1604 

Isn't the source of the problem of all of these "too early initialization" that the UiContext is created too early?

Reply:
@soosr afaik the `Wallet.Synchronizer` property (for example) stays null until `StartWalletAsync()` is called, which happens after you load a particular wallet.

This means we can't really wait for that, because we need the wallet list for the navbar among other things way before that

Without this change, multiple instances of `WalletModel` were being created for the same wallet, therefore the problem went undetected because it was re-created after wallet init, at a point where `Wallet.Synchronizer` is no longer null.

This PR attempts to avoid re-instantiation of `WalletModel` therefore it needs to deal with early creation and absence of dependencies which haven't been initialized yet.

Reply:
@ichthus1604 
After the merge of https://github.com/zkSNACKs/WalletWasabi/pull/11321 the Lazy is not needed anymore!
Could you also check in another PR if we can remove the Lazy from `IWalletCoinjoinModel` too?

Reply:
@soosr Lazy removed.

> Could you also check in another PR if we can remove the Lazy from IWalletCoinjoinModel too?

`Services.HostedServices.Get<CoinJoinManager>()` continues to crash if called early, so we still need a Lazy for that.